test/publisher/flow-publisher.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {describe, it, beforeEach, afterEach} from 'mocha';
+import * as sinon from 'sinon';
+import * as defer from 'p-defer';
+
+import {
+  Publisher,
+  flowControlDefaults,
+  PublishOptions,
+  PubsubMessage,
+} from '../../src/publisher';
+import {FlowControl} from '../../src/publisher/flow-control';
+import * as fp from '../../src/publisher/flow-publisher';
+
+class FakePublisher {
+  flowControl!: FlowControl;
+  publishMessage() {}
+  setOptions(options: PublishOptions) {
+    this.flowControl.setOptions(options.flowControlOptions!);
+  }
+}
+
+describe('Flow control publisher', () => {
+  let publisher: Publisher;
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
```

Comment:
You don't have to create the sandbox each time - you can create it once in the `before` (or just above assign it.  The `sandbox.restore()` does the resetting for you.